### PR TITLE
Feature/log in

### PR DESCRIPTION
**Pull request by tobicorradi**
## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the new behavior?

- User can now login onto the website by going to the Sign In page and filling the form.

## Other information

- Stack: React, Typescript, React Router, React Hook Form, Axios, Yup, Tailwind css and Redux Toolkit

## Preview

Screen Shot 2022-05-17 at 14 18 26

screencapture-localhost-3000-login-2022-05-17-13_56_00
screencapture-localhost-3000-login-2022-05-17-13_56_13
screencapture-localhost-3000-login-2022-05-17-13_59_00

### CODE REVIEW

**Review by natokra — APPROVED** [review]
Great job! 🚀

File: src/pages/LogIn/index.tsx [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
+  } = useForm({ resolver: yupResolver(logInSchema) });
+
+  const isRequestLoading = useMemo(() => status === RequestStatus.PENDING, [status]);
+
+  const onSubmit: SubmitHandler<LogInInputsType> = async (data: LogInInputsType) => {
+    await dispatch(logIn(data)).unwrap();
+    navigate(RoutherPaths.HOME);
+  };
+
+  useEffect(() => {
```
Comment (juanfopaez) [inline src/pages/LogIn/index.tsx:39]:
```
if(isAuthenticated) {
  ...
}
```
without useEffect should fix all redirection stuff, and you could remove line 36

File: src/pages/SignUp/index.tsx [hunk 2]
```diff
@@ -30,14 +32,17 @@ function SignUp() {
...
-      navigate(RoutherPaths.LOG_IN);
-    } catch (e) {
-      console.log(e);
-    }
+    await dispatch(signUp(data)).unwrap();
+    navigate(RoutherPaths.LOG_IN);
   };
 
+  useEffect(() => {
+    if (isAuthenticated) navigate(RoutherPaths.HOME);
```
Comment (juanfopaez) [inline src/pages/SignUp/index.tsx:40]: according to isAuthenticated comment you can move this guy out from this useEffect too :)

File: src/utils/getFormErrors.tsx [hunk 3]
```diff
@@ -8,5 +8,9 @@ export const getFormErrors = (requestErrors, errors, input) => {
     const error = errors[input].message;
     return error;
   }
+
+  if (typeof errors !== 'undefined' && typeof errors[input] === 'undefined') {
+    return requestErrors[0];
```
Comment (juanfopaez) [inline src/utils/getFormErrors.tsx:13]: 🆗  nice man :)

**Review by juanfopaez — CHANGES_REQUESTED** [review]
Good job 👍

File: src/pages/LogIn/index.tsx (hunk 1 above)
Comment (tobicorradi) [reply src/pages/LogIn/index.tsx:39]: Actually I'll get a warning from react.dev.tools if I remove the useEffect hook, it recommends you to always call navigate() inside a useEffect()

**Review by juanfopaez — APPROVED** [review]

**Comment by tobicorradi** [comment]
Thanks guys!

**Pull request merged**